src/java_bytecode/java_object_factory.cpp

```diff
@@ -863,6 +862,29 @@ void java_object_factoryt::gen_nondet_struct_init(
         true,    // allow_null always true for sub-objects
         depth,
         substruct_in_place);
+
+      if(name=="length")
+      {
+        if(class_identifier=="java.lang.String" ||
+           class_identifier=="java.lang.StringBuffer" ||
+           class_identifier=="java.lang.StringBuilder")
+        {
+          PRECONDITION(
+            object_factory_parameters.max_nondet_string_length <=
+              max_value(me.type()));
```

Comment:
Instead of a precondition here, we could just not add the assumption when this is not satisfied, as the constraint on the size will be enforced by the type anyway.
So we can replace this `PRECONDITION` by an `if`.